### ISSUE TITLE
run query requires quotes around query when not necessary

### ISSUE THREAD

**Issue by blais** [beancount/beanquery#67]
**[Original report](https://bitbucket.org/blais/beancount/issue/350) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

I have to quote the query name, otherwise I get a syntax error.  I think in both of these cases (dash and underscore) there shouldn't be a need to quote the name.

```
bean-query Financial/Ledger/books.beancount "run fundraiser_2017"
ERROR: Syntax error near '2017' (at 15)
  run fundraiser_2017
                 ^

bean-query Financial/Ledger/books.beancount "run fundraiser-2017"
ERROR: Syntax error near '-2017' (at 14)
  run fundraiser-2017
                ^
```

**Comment by dnicolodi** [comment]
The `RUN` command is parsed by beanquery as a BQL command, thus it needs to follow BQL syntax rules. In particular the query name following the `RUN` keyword is parsed as a BQL identifier. As in SQL, the identifiers cannot contain an hyphen, thus query names like `fundraiser-2017` need to be quoted:
```
run "fundraiser-2017"
```
The issue with query names like `fundraiser_2017` is not the underscore, which is permitted, but the digits: there was a bug in the parser for which digits were not permitted in identifiers. Fixed in #68.

It seems cumbersome to have to use quotes around query names. However, to avoid that the `RUN` command would need to be interpreted at a different level. Doing so in the interactive shell is no problem. Doing the same for the command line interface would require to change the syntax slightly: from
```
bean-query ledger.beans "run query_name"
```
to
```
bean-query ledger.beans run query_name
```
The second form is nicer, but it is a backward incompatible change.

**Comment by dnicolodi** [comment]
Turns out that it is possible to keep the current syntax and do not require quoting. See #69. Looking into this was a good occasion to do some cleanup in the shell implementation.
